packages/ui-toggle-details/src/ToggleDetails/index.tsx

```diff
@@ -78,8 +78,13 @@ class ToggleDetails extends Component<ToggleDetailsProps> {
     this.props.makeStyles?.({ animate: false } as ToggleDetailsStyleProps)
   }
 
-  componentDidUpdate() {
-    this.props.makeStyles?.({ animate: true } as ToggleDetailsStyleProps)
+  componentDidUpdate(prevProps: ToggleDetailsProps) {
+    // only animates the details when the expanded state changes (when ToggleDetails
+    // is used as a controlled component) - otherwise the handleToggle callback
+    // will animate the details
+    if (prevProps.expanded !== this.props.expanded) {
+      this.props.makeStyles?.({ animate: true } as ToggleDetailsStyleProps)
+    }
```

Comment:
Create an example app where changing other properties (used in styles.ts) can be tested for this logic.